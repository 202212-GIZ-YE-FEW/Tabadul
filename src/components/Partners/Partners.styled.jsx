import styled from "@emotion/styled";
import Image from "next/image";

export const Header = styled.h1`
    text-align: center;
    font-style: normal;
    font-weight: 700;
    font-size: 56px;
    line-height: 84px;
    color: #3c4347;
    margin-bottom: 90px;
    @media screen and (max-width: 768px) {
        font-size: 36px;
    }
`;

export const Container = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    margin-bottom: 152.5px;
    gap: 180px;
    @media screen and (max-width: 768px) {
        flex-direction: column;
        gap: 50px;
    }
`;

export const PartnerImage = styled(Image)`
    display: flex;
    justify-contnet: center;
    align-items: center;
`;
