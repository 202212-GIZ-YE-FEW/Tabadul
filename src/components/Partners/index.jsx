import React from "react";
import { Navigation, Pagination } from "swiper";
import { Swiper, SwiperSlide } from "swiper/react";
import "swiper/css/navigation";
import "swiper/css/pagination";
import "swiper/css";

import {
    Container,
    Header,
    PartnerImage,
    PartnersContainer,
} from "./Partners.styled";
import sevenUp from "../../assets/Image/sevenup.svg";
import recoded from "../../assets/Image/recoded.png";
import shareshop from "../../assets/Image/share1.png";
import shareshoptwo from "../../assets/Image/share2.png";
import amazon from "../../assets/Image/amazon.png";
import pebble from "../../assets/Image/pebble.png";
function Partners() {
    return (
        <PartnersContainer>
            <Header>Our Partners</Header>
            <Container>
                <Swiper
                    breakpoints={{
                        0: {
                            slidesPerView: 1,
                        },
                        400: {
                            slidesPerView: 1,
                        },
                        639: {
                            slidesPerView: 2,
                        },
                        865: {
                            slidesPerView: 2,
                        },
                        1000: {
                            slidesPerView: 3,
                        },
                        1500: {
                            slidesPerView: 3,
                        },
                        1700: {
                            slidesPerView: 4,
                        },
                    }}
                    modules={[Navigation, Pagination]}
                    spaceBetween={50}
                    slidesPerView={3}
                    navigation
                    pagination={{ clickable: true }}
                >
                    <SwiperSlide>
                        <PartnerImage src={sevenUp} />
                    </SwiperSlide>
                    <SwiperSlide>
                        <PartnerImage src={recoded} />
                    </SwiperSlide>
                    <SwiperSlide>
                        <PartnerImage src={shareshop} />
                    </SwiperSlide>
                    <SwiperSlide>
                        <PartnerImage src={pebble} />
                    </SwiperSlide>
                    <SwiperSlide>
                        <PartnerImage src={shareshoptwo} />
                    </SwiperSlide>
                    <SwiperSlide>
                        <PartnerImage src={amazon} />
                    </SwiperSlide>
                </Swiper>
            </Container>
        </PartnersContainer>
    );
}

export default Partners;
