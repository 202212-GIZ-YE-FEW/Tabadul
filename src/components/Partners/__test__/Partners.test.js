import renderer from "react-test-renderer";

import Partners from "..";

it("renders correctly", () => {
    const tree = renderer.create(<Partners />).toJSON();
    expect(tree).toMatchSnapshot();
});
