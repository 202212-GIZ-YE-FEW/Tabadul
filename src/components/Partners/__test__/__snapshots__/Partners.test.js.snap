// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="css-127jica"
  >
    Our Partners
  </h1>,
  <div
    className="css-6j60lp"
  >
    <img
      alt="Partner"
      className="css-0"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={40}
    />
    <img
      alt="Partner"
      className="css-0"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={40}
    />
    <img
      alt="Partner"
      className="css-0"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={40}
    />
    <img
      alt="Partner"
      className="css-0"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={40}
    />
  </div>,
]
`;
